[PE-D][Tester C] `find` command attempts to search for empty strings instead of validating missing arguments

**Steps to reproduce:**

1. Launch the ItemTasker application. 
2. Execute the `find` command with flags present but no values provided (e.g., `find n/ t/ i/` or `find n/`). 

**Expected Behaviour:** The application should validate the contents of the arguments, not just the presence of the flags. If the provided values are empty strings, it should effectively treat them as missing and throw a usage error (e.g., "At least one valid filter value must be provided").

**Actual behavior:** The system accepts the empty string as a valid search parameter. It prioritizes the `n/` flag and attempts to look up an SKU with a blank ID, resulting in a misleading `SKUNotFoundException` (`[ERROR] SKU '' not found. Use 'listtasks' to see available SKUs.`).

<img width="1350" height="196" alt="Image" src="https://raw.githubusercontent.com/NUS-CS2113-AY2526-S2/ped/master/files/1655beb4-b638-4c45-96f6-f259520bb7a7.png" />

-------------
Labels: `severity.Low` `type.FunctionalityBug`

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -223,19 +223,13 @@ Here's a class diagram of the SKUTask component:
 
 ![SKUTask Architecture Diagram](diagrams/skutask-operations/skutask-architecture.png)
 
-The sequence diagram below illustrates the interactions within the SKUTask component, taking the `addSKUTask` API call as an example.
+The `SKUTask` component operates on a principle of **Strict Delegation**. The `SKUTaskList` acts as a facade for the `SKUTask` objects it contains, ensuring that higher-level components (like `TaskCommandHandler`) never mutate a task directly.
 
-**Interactions Inside the SKUTask Component for the `addskutask` Command**
+* **Encapsulation:** All task properties (Date, Priority, Description) are modified via wrapper methods in `SKUTaskList`.
+* **State Mutation:** Interactions permanently mutate the memory model, which is then handled by the `Storage` component upon exit.
 
-![Add SKU Task Sequence Diagram](diagrams/skutask-operations/addTaskSequence.png)
-
-How the SKUTask component works:
-
-When the `SKUTaskList` is called upon to execute a task-level operation, it receives the extracted task properties (e.g., date, priority, description) from the parent `SKU` that delegates it.
-This results in a `SKUTask` object being instantiated (or an existing one being modified or deleted) which is managed entirely within the `SKUTaskList`.
-The `SKUTaskList` communicates internally with the individual `SKUTask` items, enforcing controlled access through wrapper methods.
-Note that although this is shown as a single step in the diagram above (for simplicity), in the code it can take several interactions (between the list and the individual task fields) to update specific parameters like the `Priority` enum or completion status.
-The result of the task execution permanently mutates the system's memory model, which is ultimately fetched back by higher-level components for confirmation.
+> [!NOTE]
+> For the detailed end-to-end sequence of how a task is added from a user command down to this component, please refer to the [Implementation: Add SKU Task](#add--delete-sku-task-feature) section.
 
 Here are the other interactions in the SKUTask component (omitted from the sequence diagram above) that are used for property reading and modifications:
 

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -207,6 +207,13 @@ Example of usage:
 
 `listtasks l/B2`
 
+#### Handling Multiple Filters:
+
+If more than one filter is entered, the system returns an error message to indicate the command is invalid.
+
+* **Example Command:** `listtasks n/PALLET-A p/HIGH`
+* **Expected Output:** `[ERROR] Conflict: You can only use ONE filter (n/, p/, or l/) at a time.`
+
 ---
 
 ### Finding tasks: `find`
@@ -230,9 +237,11 @@ Example of usage:
 ---
 
 ### Viewing warehouse map: `viewmap`
-Displays a visual representation of the 3×3 warehouse grid showing which locations are occupied.
+Displays a visual representation of the 3×3 warehouse grid.
 
-Format: `viewmap`
+**Map Logic:**
+* **What the numbers mean:** Each cell (e.g., `[B2: 05]`) displays the **total number of tasks** (both completed and incomplete) assigned to all SKUs at that location.
+* **Why a location might show `00`:** If a location shows `00`, it means there are no tasks created for the SKUs at that spot. It does **not** mean the location is empty of SKUs.
 
 Example of usage:
 
@@ -316,6 +325,9 @@ Example of usage:
 
 **Q**: Can I edit multiple fields of a task at once?  
 **A**: Yes. The `edittask` command accepts any combination of `d/` (due date), `p/` (priority), and `t/` (description) flags simultaneously. Only the fields you specify will be updated. All other fields are preserved.
+
+**Q**: Can I have more than one SKU in the same grid square (e.g., A1)?  
+**A**: Yes. ItemTasker treats grid locations as "sectors." You can store multiple different SKUs in the same sector. The `viewmap` will show the sum of all tasks for all SKUs in that specific sector.
 
 ## Command Summary
 

--- a/src/main/java/command/ViewCommandHandler.java
+++ b/src/main/java/command/ViewCommandHandler.java
@@ -233,6 +233,10 @@ public class ViewCommandHandler {
         String descFilter = cmd.getArg("t");
         String indexStr = cmd.getArg("i");
 
+        if (skuFilter != null && skuFilter.trim().isEmpty()) {
+            throw new InvalidFilterException("The SKU keyword after n/ cannot be empty.");
+        }
+
         if (descFilter != null && descFilter.trim().isEmpty()) {
             throw new InvalidFilterException("The description keyword after t/ cannot be empty.");
         }

--- a/src/main/java/skutask/ViewSKUTask.java
+++ b/src/main/java/skutask/ViewSKUTask.java
@@ -101,10 +101,11 @@ public class ViewSKUTask {
             return 0;
         }
         String targetName = parentSku.getSKULocation().name();
+        String currentPosUpper = currentPos.toUpperCase();
         int targetX = targetName.charAt(0) - 'A';
         int targetY = Character.getNumericValue(targetName.charAt(1)) - 1;
-        int currentX = currentPos.toUpperCase().charAt(0) - 'A';
-        int currentY = Character.getNumericValue(currentPos.charAt(1)) - 1;
+        int currentX = currentPosUpper.charAt(0) - 'A';
+        int currentY = Character.getNumericValue(currentPosUpper.charAt(1)) - 1;
         return Math.abs(targetX - currentX) + Math.abs(targetY - currentY);
     }
 }

--- a/src/main/java/ui/Parser.java
+++ b/src/main/java/ui/Parser.java
@@ -56,15 +56,20 @@ public class Parser {
 
         for (String token : tokens) {
             int slashIdx = token.indexOf('/');
-            if (slashIdx > 0) {
-                String key = token.substring(0, slashIdx).trim().toLowerCase();
-                String value = token.substring(slashIdx + 1).trim();
-                if (args.containsKey(key)) {
-                    throw new InvalidCommandException(
-                            "Duplicate flag '" + key + "/' is not allowed.");
-                }
-                args.put(key, value);
+
+            if (slashIdx <= 0) {
+                throw new InvalidCommandException(
+                        "Invalid argument format: '" + token + "'. " +
+                                "Make sure to use a prefix (e.g., n/, p/, l/)."
+                );
             }
+            String key = token.substring(0, slashIdx).trim().toLowerCase();
+            String value = token.substring(slashIdx + 1).trim();
+            if (args.containsKey(key)) {
+                throw new InvalidCommandException(
+                        "Duplicate flag '" + key + "/' is not allowed.");
+            }
+            args.put(key, value);
         }
 
         return args;

